05-specifying-sample-designs.Rmd

```diff
@@ -1 +1,673 @@
+
+```{r}
+#| label: acsexamp
+#| cache: TRUE
+#| results: 'hide'
+library(tidycensus)
+
+pums_in <- get_pums(variables=c("NP", "BDSP", "HINCP"), state="37", 
+                    puma=c("01301", "01302"), rep_weights = "housing", 
+                    year=2020, survey="acs5",
+                    variables_filter=list(SPORDER=1, TYPEHUGQ=1))
+
+
+```
+
+```{r}
+#| label: acsexampcont
+#| cache: TRUE
+#| dependson: 'acsexamp'
+pums_in
+
+des_acs <- pums_in %>%
+    as_survey_rep(weights = WGTP, repweights= num_range("WGTP", 1:80),
+                type="JK1", mse=TRUE, scale=4/80)
+
+des_acs
+
+summary(des_acs)
+```
+
+
+### Bootstrap method
+
+In bootstrap resampling, replicates are created by selecting random samples of the PSUs with replacement. If there are $M$ PSUs in the sample, then each replicate will be created by selecting a random sample of $M$ PSUs with replacement. Each replicate is created independently and the weights for each replicate are adjusted to to reflect the population generally using the same method as the analysis weight was adjusted. 
+
+#### The math {-}
+
+A weighted estimate for the full sample is calculated as $\hat{\theta}$ and then a weighted estimate for each replicate is calculated as $\hat{\theta}_r$ for $R$ replicates. Then the standard error of the estimate is calculated as:
+
+$$se(\hat{\theta})=\sqrt{\alpha \sum_{r=1}^R \left( \hat{\theta}_r-\hat{\theta}\right)^2}$$
+
+#### The syntax {-}
+
+If a dataset had WT0 for the main weight, had 20 bootstrap weights indicated WT1, WT2, ..., WT20, and $\alpha=.02$, use the following syntax:
+
+```r
+des_bs <- dat %>%
+  as_survey_rep(weights = WT0, repweights= num_range("WT", 1:20),
+                type="bootstrap", mse=TRUE, scale=.02)
+
+```
+
+Note that the scale is usually provided in documentation and is a constant so is not provided as a variable in the tibble.
+
+#### Example {-}
+
+Returning to the api example, bootstrap weights were constructed for a one cluster design. 50 replicate weights were created on a dataset `apiclus1_slim` which has some familiar variables including cds, dnum, fpc, and pw but now additionally includes bootstrap weights pw1, ..., pw50. The scale $(\alpha)$ is $15/(14*49)=0.02186589$
+
+```{r}
+#| label: genbs
+#| include: FALSE
```

Comment:
I just realized that you have this as `include=FALSE`, so maybe some of my other comments are not as relevant.  However, let's discuss how much detail about the data prep we're doing should be included, both code wise and text wise.

Reply:
Yes, that was too messy to include but wanted to generate an example.